File: Java/0844-backspace-string-compare.java
class Solution {
    public String removeBackspace(String s){
        Stack<Character> ans = new Stack();
        for(char c: s.toCharArray()){
            if( c != '#')
                ans.push(c);
            else if( ! ans.empty())
                ans.pop();
        }
        return String.valueOf(ans);
        }
    public boolean backspaceCompare(String s, String t) {
        return removeBackspace(s).equals(removeBackspace(t));   
    }
}

File: Python/0844-backspace-string-compare.py
class Solution:
    def removeBackspace(self, s: str):
        ans = []
        for c in s:
            if c != '#':
                ans.append(c)
            elif ans:
                ans.pop()
        return "".join(ans)
    def backspaceCompare(self, s: str, t: str) -> bool:
        return self.removeBackspace(s) == self.removeBackspace(t)